### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by djacu**
Adds CI for testing the devshell templates.

Okay I have good news and bad news.
Good news: It works!
Bad news: It takes a while...

### CODE REVIEW

**Comment by rapenne-s** [comment]
maybe it would be a lot faster if you use nickel cachix substituter? Because otherwise you have to compile nickel every time, which is wrong

**Comment by djacu** [comment]
@rapenne-s you were totally right. It's much faster now with the latest commit.

File: templates/devshells/c/flake.nix [hunk 1]
```diff
@@ -5,6 +5,11 @@
   inputs.flake-utils.url = "github:numtide/flake-utils";
   inputs.nickel-nix.url = "github:nickel-lang/nickel-nix";
 
+  nixConfig = {
+    extra-substituters = ["https://tweag-nickel.cachix.org"];
+    extra-trusted-public-keys = ["tweag-nickel.cachix.org-1:GIthuiK4LRgnW64ALYEoioVUQBWs0jexyoYVeLDBwRA="];
+  };
+
```
Comment (rapenne-s) [inline templates/devshells/c/flake.nix:12]: This comes from the template directory generating the templates directories, you should commit the original file too

File: .github/workflows/templates.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    - name: Initialize devshell ${{ matrix.template }}
+      run: |
+        nix flake new --template path:$PWD/repo#${{ matrix.template }} example --accept-flake-config
+
+    - name: Enter devshell ${{ matrix.template }}
+      run: |
+          pushd ./example
+          nix develop --impure --accept-flake-config
+          popd
```
[the remark this answers is not in the document]
Comment (djacu) [reply .github/workflows/templates.yml:67]: I believe it is working. The CI used to take 10-15 minutes before adding the substituters, and now it takes 1 minute. From looking at the CI logs, it appears to do what I think it should to.

File: templates/devshells/c/flake.nix (hunk 1 above)
Comment (rapenne-s) [reply templates/devshells/c/flake.nix:12]: nevermind, I thought we still had a `rust-devshell` directory where development is happening, and the `rust` example that was created from it :sweat_smile:

**Pull request merged**